tests/tests/test_github_webhooks.py

```diff
@@ -0,0 +1,605 @@
+# Copyright 2021 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import os
+import requests
+import time
+
+BASE_DIR = os.path.dirname(__file__)
+
+
+def load_payload(filename):
+    with open(os.path.join(BASE_DIR, "payloads", filename), "rb") as f:
+        return f.read()
+
+
+def test_pull_request_opened(integration_test_runner_url):
+    res = requests.post(
+        integration_test_runner_url + "/",
+        data=load_payload("pull_request_opened.json"),
+        headers={
+            "Content-Type": "application/json",
+            "X-Github-Event": "pull_request",
+            "X-Github-Delivery": "delivery",
+        },
+    )
+    assert res.status_code == 200
+    #
+    res = requests.get(integration_test_runner_url + "/logs",)
+    assert res.status_code == 200
+    assert res.json() == [
+        "git.Run: /usr/bin/git init .",
+        "git.Run: /usr/bin/git remote add github git@github.com:/mendersoftware/workflows.git",
+        "git.Run: /usr/bin/git remote add gitlab git@gitlab.com:Northern.tech/Mender/workflows",
+        "git.Run: /usr/bin/git fetch github pull/140/head:pr_140",
+        "git.Run: /usr/bin/git push -f --set-upstream gitlab pr_140",
+        "info:Created branch: workflows:pr_140",
+        "info:Pipeline is expected to start automatically",
+        "debug:deleteStaleGitlabPRBranch: PR not closed, therefore not stopping it's pipeline",
+        "info:Ignoring cherry-pick suggestions for action: opened, merged: false",
+        "debug:stopBuildsOfStalePRs: PR not closed, therefore not stopping it's pipeline",
+        "info:Pull request event with action: opened",
+        "git.Run: /usr/bin/git pull --rebase origin",
+        "info:workflows/master is being used in the following integration: [master]",
+        "info:the following integration branches: [master] are using workflows/master",
+        "info:workflows:140 would trigger 1 builds",
+        "info:I have already commented on the pr: workflows/140, no need to keep on nagging",
+    ]
```

Comment:
This is just amazing, really.

A part from covering it's main purpose (check regressions) it also is showing us where we can improve things (for example here we are attempting to stop the pipeline twice).